client/src/components/Editor.jsx

```diff
@@ -184,50 +193,62 @@ export default function Editor({schema, endpoint, setQuery}) {
 
   /****************************************** Helper Functions ********************************************/
 
-    /* Execute Current Operation in Query Pane (cmd + enter OR auto) */
+  /* Get operations pane content and checks its validation */
+  const getOperationsAndValidate = () => {
+    const operations = editor
+      .getModel(Uri.file('operation.graphql'))
+      .getValue();
+    if (!validateBrackets(operations)) {
+      alert('Invalid brackets'); // TODO: refactor error handling
+      return;
+    }
+    if (operations.trim() === '') {
+      alert('Empty query'); // TODO: refactor error handling
+      return;
+    }
+    return operations;
+  };
+
+  /* Execute Current Operation in Query Pane (cmd + enter OR auto) */
   const execOperation = async function () {
     console.log('HERELOL');
     if (!currentSchema.current) {
       alert('Please load a valid schema'); // TODO: refactor error handling
       return;
     }
-    const markers = editor.getModelMarkers({resource: Uri.file('operation.graphql')});
+    const markers = editor.getModelMarkers({
+      resource: Uri.file('operation.graphql'),
+    });
     if (markers.length) {
       alert('Syntax error :)'); // TODO: refactor error handling
       return;
     }
     // Grab the code from the variables pane
     const variables = editor.getModel(Uri.file('variables.json')).getValue();
     // Grab the operations from the operations pane
-    const operations = editor.getModel(Uri.file('operation.graphql')).getValue();
-    if (!validateBrackets(operations)) {
-      alert('Invalid brackets'); // TODO: refactor error handling
-      return;
-    };
-    if (operations.trim() === '') {
-      alert('Empty query'); // TODO: refactor error handling
-      return;
-    }
+    const validOperations = getOperationsAndValidate();
```

Comment:
Potential problem here because the getOperationsAndValidate() function short circuits itself with returns but does not cause the parent caller to return, therefore, there may be cases where you attempt to validate things, and get a valid alert telling you the errors, however execution still goes through and you can get console errors (unwanted kind) or a crash

Reply:
The way it was before, not being in a separate function made the returns exit out of execOperation() -- but with this refactor you will probably need to return something explicit like null to short circuit and catch it in execOperation in order to properly short circuit there before fetching

Reply:
Absolutely right! Yeah, if it is changed to return null and then checked if it's falsy to return, then this bug should be squashed, correct?

Reply:
Or would we rather just leave the code as it was before the refactor. Not a problem to do, just wanted thought having that helper func there made things a little tidier